### ISSUE TITLE
Improve Python API

### ISSUE THREAD

**Issue by verhovsky** [adrienverge/yamllint#707]
I'm using yamllint from Python because my YAML is frontmatter in Markdown files, so I have to parse it out before linting it and I have to say the API is bad/unpythonic. I would expect this

```py
import yamllint

yamllint.lint("foo: bar")
```

to work with the default config, and I should not be required to make a YAMLConfig object (from a string!), I should be able to just pass a dictionary

```
import yamllint

yamllint.lint("foo: bar", "relaxed")

yamllint.lint("foo: bar", {"extends": "relaxed", "rules": {}})
```

also, https://yamllint.readthedocs.io/en/stable/development.html is still outdated after #699 which wasted some of my time. Also the readme should mention that there is a Python API, because I couldn't tell.
